### DISCUSSION TITLE
is there a way to remove the ?callbackUrl injected from next auth after signing out?

### DISCUSSION THREAD

**Discussion by renanleonel** [vercel/next.js#59446] [Help]
### Summary

hi! im trying to set up my authentication flow following the [Next.js Learn](https://nextjs.org/learn/dashboard-app/adding-authentication) tutorial. currently i'm facing some issues with the authentication using middlewares (next auth v5).

currently the sign in is working fine, but when i sign out or try to access a protected folder without being authenticated, next auth injects on my url this param:
http://localhost:3000/?callbackUrl=http%3A%2F%2Flocalhost%3A3000%2Fdashboard

i would like to just redirect the user to the ' / ' path, without the params. is there a way to do this?

my code is almost the exact the one that vercel provides in the [Next.js Learn: Adding Authentication](https://nextjs.org/learn/dashboard-app/adding-authentication) 

thank you!

### Example

https://nextjs.org/learn/dashboard-app/adding-authentication

**Comment by Naidev7** [comment]
Hey @renanleonel , I have the same issue in my project. Do you find any solution to solve this??

**Reply by renanleonel** [reply]
i'm currently using another approach, using middleware. with this, i'm not facing any issues with the callbackUrl. i have an example in [this repo](https://github.com/renanleonel/next-auth-v5-middleware)
